EU4toV2/Source/EU4World/Provinces/Provinces.cpp

```diff
@@ -1,35 +1,40 @@
-#include "Log.h"
 #include "Provinces.h"
+#include "Log.h"
 #include "ParserHelpers.h"
 #include <fstream>
 
 EU4::Provinces::Provinces(std::istream& theStream)
 {
-	registerRegex("-[0-9]+", [this](const std::string& numberString, std::istream& theStream)
-	{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void EU4::Provinces::registerKeys()
+{
+	registerRegex("-[0-9]+", [this](const std::string& numberString, std::istream& theStream) {
 		auto newProvince = std::make_shared<Province>(numberString, theStream);
 		provinces.insert(std::make_pair(newProvince->getNum(), std::move(newProvince)));
 	});
-	registerRegex("[a-zA-Z0-9_\\.:]+", commonItems::ignoreItem);
-
-	parseStream(theStream);
-	clearRegisteredKeywords();
+	registerRegex(commonItems::catchallRegex, commonItems::ignoreItem);
 }
 
-std::shared_ptr<EU4::Province> EU4::Provinces::getProvince(const int provinceNumber)
+const std::shared_ptr<EU4::Province>& EU4::Provinces::getProvince(const int provinceNumber) const
```

Comment:
Can't the return type be auto?

Reply:
![image](https://user-images.githubusercontent.com/2392697/99077071-7b70c700-25bc-11eb-9494-13e4b22ef6c5.png)
![image](https://user-images.githubusercontent.com/2392697/99077115-8b88a680-25bc-11eb-8e2e-b4fbbe869b44.png)
You can, but it's fuzzy about having not being defined although it obviously is, and I don't see immediately what's the return type when looking at the cpp file.

Reply:
On the screenshot, did you also change it to auto in the definition?

Reply:
That's fine, auto is mostly for our convenience anyway.